bson/benchmark_test.go

```diff
@@ -121,14 +127,174 @@ var nestedInstance = nestedtest1{
 	},
 }
 
-func BenchmarkEncoding(b *testing.B) {
-	for i := 0; i < b.N; i++ {
-		_, _ = Marshal(encodetestInstance)
+const extendedBSONDir = "../data/extended_bson"
+
+func mustReadBSONTestData(filename string) map[string]interface{} {
+	filePath := path.Join(extendedBSONDir, filename)
+	file, err := os.Open(filePath)
+	if err != nil {
+		panic(fmt.Sprintf("error opening file %q: %s", filePath, err))
+	}
+	defer file.Close()
+
+	gz, err := gzip.NewReader(file)
+	if err != nil {
+		panic(fmt.Sprintf("error creating GZIP reader: %s", err))
+	}
+	defer gz.Close()
+
+	data, err := ioutil.ReadAll(gz)
+	if err != nil {
+		panic(fmt.Sprintf("error reading GZIP contents of file: %s", err))
+	}
+
+	var v map[string]interface{}
+	err = UnmarshalExtJSON(data, false, &v)
+	if err != nil {
+		panic(fmt.Sprintf("error unmarshalling extended JSON: %s", err))
+	}
+
+	return v
+}
+
+func BenchmarkMarshal(b *testing.B) {
```

Comment:
To clarify, for the `gz` files, `BenchmarkMarshal` `Unmarshal`s and then `Marshal`s, and `BenchmarkUnmarshal` `Unmarshal`s, `Marshal`s and then `Unmarshal`s again? The names of the benchmarks are a little confusing, and I wonder if you need both.

Reply:
Maybe add a comment to `mustReadBSONTestData`? Which I think could optionally be `readBSONTestData` instead.

Reply:
Renamed and added a comment to `mustReadBSONTestData`.